libmachine/kv/kv.go

```diff
@@ -0,0 +1,101 @@
+package kv
+
+// libkv layer for machine
+// holds globals and initial functions for libkv usage
+// put this here to avoid import loops in other code areas
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"time"
+
+	"github.com/docker/libkv"
+	"github.com/docker/libkv/store"
+	"github.com/docker/libkv/store/etcd"
+	"github.com/docker/machine/libmachine/log"
+)
+
+const MachineKvPrefix = "machine/v0"
+
+var kvStore store.Store
+
+func Connect(kvHost string) (err error) {
+	log.Debugf("AK: connect %d: %s", os.Getpid(), kvHost)
+	// TODO - figure out how to get TLS support in here...
+	if kvStore != nil {
+		log.Debugf("AK: redundant connect, ignoring")
+		return nil
+	}
+	etcd.Register()
+	kvStore, err = libkv.NewStore(store.ETCD,
+		[]string{kvHost},
+		&store.Config{
+			ConnectionTimeout: 10 * time.Second,
+		},
+	)
+
+	if err != nil {
+		log.Warnf("AK: err in connect, this is BAD: %s", err)
+	}
+
+	return err
+}
+
+func KvList(dir string) (kvList []*store.KVPair, err error) {
+	log.Debugf("AK: list %d", os.Getpid())
+	if kvStore == nil {
+		panic(fmt.Errorf("KVStore not initialized!!"))
```

Comment:
Ultimately we should clean up the call sites so this should never happen, and return an error instead of panic for library usage.